docs/architecture/agent-in-depth.md

```diff
@@ -38,3 +38,64 @@ Supported Rust DHs each have a [library](https://github.com/project-akri/akri/tr
 
 Reference the [Discovery Handler development document](../development/handler-development.md) to learn how to implement a Discovery Handler.
 
+## Passing additional properties to Discovery Handlers
+
+In addition to the `discoveryDetails` in Configuration that sets details for narrowing the Discovery Handlers' search, the `discoveryProperties` can be used to pass additional information to Discovery Handler. One of scenarios that can leverage `discoveryProperties` is to pass credential data to Discovery Handlers to perform
+authenticated resource discovery.
+It's not uncommon that resources that a Configuration requests to discovery are in 'authenticated-enabled' mode that a Discovery Handler can discover the
+existances of resources but requires credentials to authenticate access details information of the resource. The credential data can be placed in 
```

Comment:
This sentence feels overly complicated, maybe something like this would be better:

> It is common for a device to require authentication in order to access its properties. The Discovery Handler then need these credentials to properly discover and filter the device. The credential data ...

Reply:
update the sentence